extract image in textnode

unsleepy22 opened the pull request:
fix the issue that images within text nodes are not extracted, which is not unusual in web pages. 
i.e.,
```
<p><img src="aa.jpg"></img><p>
```

adbar commented on trafilatura/utils.py:
```diff
@@ -346,6 +346,20 @@ def trim(string: str) -> str:
         return ""
 
 
+def is_image_element(element: _Element):
```
The return type should be `-> bool` right?

unsleepy22 replied:
ok, fixed

adbar commented on trafilatura/htmlprocessing.py:
```diff
@@ -226,6 +226,8 @@ def handle_textnode(
     preserve_spaces: bool = False,
 ) -> Optional[_Element]:
     "Convert, format, and probe potential text elements."
+    if is_image_element(elem):
```
This will slow down processing, is it necessary?

Suggestion:
`if elem.tag == "graphic" and is_image_element(elem):`

unsleepy22 replied:
ok, fixed

adbar commented:
Hi @unsleepy22, thanks for the PR. I still have questions about it (see above), could you please have a look?

unsleepy22 commented:
> Hi @unsleepy22, thanks for the PR. I still have questions about it (see above), could you please have a look?

thanks for your comments, fixed and rebased my commits, would you take a look again?

adbar commented:
LGTM, thanks.
